src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -337,12 +342,21 @@ Copyright (c) .NET Foundation. All rights reserved.
 
     <!-- Get items from this project last so that they will be copied last. -->
     <ItemGroup>
+      <_ContentItemsToPublish Include="@(Content->'%(FullPath)')"
+                              Condition="'%(Content.CopyToPublishDirectory)'=='Always' or '%(Content.CopyToPublishDirectory)'=='PreserveNewest'"/>
+    </ItemGroup>
+
+    <AssignTargetPath Files="@(_ContentItemsToPublish)" RootFolder="$(MSBuildProjectDirectory)">
+      <Output TaskParameter="AssignedFiles" ItemName="_ContentItemsToPublishWithTargetPath" />
+    </AssignTargetPath>
+    
+    <ItemGroup>
       <_SourceItemsToCopyToPublishDirectoryAlways KeepMetadata="$(_GCTPDIKeepMetadata)"
-                                                  Include="@(ContentWithTargetPath->'%(FullPath)')"
-                                                  Condition="'%(ContentWithTargetPath.CopyToPublishDirectory)'=='Always'"/>
```

Comment:
It looks like `ContentWithTargetPath` is used to resolve items that should be published from referenced projects, and hence this change is breaking that and some tests that cover this are failing.